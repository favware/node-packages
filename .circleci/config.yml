version: 2.1

orbs:
  codecov: codecov/codecov@1.0.5

aliases:
  - &restore_yarn_cache
    restore_cache:
      name: Restore Yarn Package Cache
      keys:
        - yarn-packages-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}
        - yarn-packages-{{ arch }}-{{ .Branch }}-
        - yarn-packages-{{ arch }}-
  - &run_yarn
    run:
      name: Install Packages
      command: yarn --frozen-lockfile
  - &store_artifacts
    store_artifacts:
      path: '**/coverage'
  - &codecov
    codecov/upload:
      file: '**/coverage/*.json'
  - &checkout
    checkout

commands:

  setup:
    description: Initial setup for favware projects
    steps:
      - *checkout
      - *restore_yarn_cache
      - *run_yarn
      - save_cache:
          name: Save Yarn Package Cache
          key: yarn-packages-{{ arch }}-{{ .Branch }}-{{ checksum "yarn.lock" }}
          paths:
            - ~/.cache/yarn
            - node_modules

  lint:
    description: Lints Favware projects
    steps:
      - *checkout
      - *restore_yarn_cache
      - *run_yarn
      - run:
          name: Lint all libraries
          command: yarn lint

  test:
    description: Tests Favware projects
    steps:
      - *checkout
      - *restore_yarn_cache
      - *run_yarn
      - run:
          name: Test all libraries
          command: yarn test:coverage

  build:
    description: Builds Favware projects
    steps:
      - *checkout
      - *restore_yarn_cache
      - *run_yarn
      - run:
          name: Build packages
          command: yarn build
      - persist_to_workspace:
          root: build
          paths:
            - node_modules
            - dist

  process_artifacts:
    description: Process artifacts for Favware projects
    steps:
      - *checkout
      - *restore_yarn_cache
      - *run_yarn
      - *store_artifacts
      - *codecov

# List of jobs
jobs:
  node-v10:
    docker:
      - image: circleci/node:10
    steps:
      - setup
      - lint
      - test
      - build
      - process_artifacts
  node-v11:
    docker:
      - image: circleci/node:11
    steps:
      - setup
      - lint
      - test
      - build
      - process_artifacts
  node-v12:
    docker:
      - image: circleci/node:12
    steps:
      - setup
      - lint
      - test
      - build
      - process_artifacts

workflows:
  version: 2
  node-multi-build:
    jobs:
      - node-v10
      - node-v11
      - node-v12

  every-sunday-at-night:
    triggers:
      - schedule:
          cron: "0 0 * * 0"
          filters:
            branches:
              only:
                - master
    jobs:
      - node-v10
      - node-v11
      - node-v12