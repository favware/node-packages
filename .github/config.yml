pull_request_rules:
  - name: Automatic Depfu merging
    conditions:
      - author=depfu[bot]
      - approved-reviews-by=Favna
      - base=main
      - label~=^Meta:\ Dependencies
    actions:
      merge:
        method: squash
