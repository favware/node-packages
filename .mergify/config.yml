pull_request_rules:
  - name: Automatic Depfu merging
    conditions:
      - approved-reviews-by=Favna
      - base=main
      - label~=^Meta:\ Dependencies
      - title~=Yarn\ dependencies
    actions:
      merge:
        method: squash
