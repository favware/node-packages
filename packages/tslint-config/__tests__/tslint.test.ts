import config from '../src';

describe('TSLint Config', () => {
  test('should export config', () => {
    expect(config.rules['arrow-return-shorthand']).toBe(true);
    expect(config).toMatchSnapshot();
  });
});