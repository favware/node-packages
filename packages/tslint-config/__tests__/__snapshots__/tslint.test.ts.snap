// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TSLint Config should export config 1`] = `
Object {
  "rules": Object {
    "align": Array [
      true,
      "parameters",
      "statements",
      "members",
      "elements",
    ],
    "array-type": Array [
      true,
      "array",
    ],
    "arrow-parens": Array [
      true,
      "ban-single-arg-parens",
    ],
    "arrow-return-shorthand": true,
    "await-promise": true,
    "ban-comma-operator": true,
    "binary-expression-operand-order": true,
    "class-name": true,
    "comment-format": Array [
      true,
      "check-space",
      "check-uppercase",
    ],
    "curly": Array [
      true,
      "ignore-same-line",
    ],
    "cyclomatic-complexity": Array [
      true,
      25,
    ],
    "encoding": true,
    "eofline": false,
    "forin": false,
    "import-spacing": true,
    "indent": Array [
      true,
      "spaces",
      2,
    ],
    "interface-over-type-literal": false,
    "label-position": true,
    "linebreak-style": Array [
      true,
      "LF",
    ],
    "max-classes-per-file": Array [
      true,
      1,
    ],
    "max-line-length": Array [
      true,
      200,
    ],
    "member-ordering": Array [
      true,
      Object {
        "order": "fields-first",
      },
    ],
    "newline-before-return": false,
    "no-angle-bracket-type-assertion": true,
    "no-arg": true,
    "no-bitwise": false,
    "no-boolean-literal-compare": true,
    "no-conditional-assignment": true,
    "no-consecutive-blank-lines": Array [
      true,
      2,
    ],
    "no-console": Array [
      true,
      "log",
      "error",
      "group",
      "groupEnd",
      "groupCollapsed",
      "table",
      "trace",
    ],
    "no-construct": true,
    "no-debugger": true,
    "no-duplicate-imports": true,
    "no-duplicate-super": true,
    "no-duplicate-switch-case": true,
    "no-duplicate-variable": true,
    "no-dynamic-delete": true,
    "no-empty": true,
    "no-empty-interface": true,
    "no-eval": true,
    "no-implicit-dependencies": false,
    "no-invalid-template-strings": true,
    "no-invalid-this": true,
    "no-misused-new": true,
    "no-null-keyword": false,
    "no-object-literal-type-assertion": true,
    "no-reference": true,
    "no-require-imports": false,
    "no-return-await": true,
    "no-shadowed-variable": true,
    "no-sparse-arrays": true,
    "no-string-literal": true,
    "no-string-throw": true,
    "no-tautology-expression": true,
    "no-this-assignment": true,
    "no-trailing-whitespace": Array [
      true,
      "ignore-jsdoc",
    ],
    "no-unsafe-finally": true,
    "no-unused-expression": true,
    "no-var-keyword": true,
    "no-var-requires": true,
    "number-literal-format": true,
    "object-literal-key-quotes": Array [
      true,
      "as-needed",
    ],
    "object-literal-shorthand": Array [
      true,
    ],
    "object-literal-sort-keys": Array [
      true,
      "match-declaration-order-only",
    ],
    "one-line": Array [
      true,
      "check-open-brace",
      "check-catch",
      "check-else",
      "check-whitespace",
    ],
    "only-arrow-functions": true,
    "ordered-imports": true,
    "prefer-conditional-expression": Array [
      true,
      "check-else-if",
    ],
    "prefer-const": Array [
      true,
      Object {
        "destructuring": "all",
      },
    ],
    "prefer-for-of": true,
    "prefer-method-signature": true,
    "prefer-object-spread": true,
    "prefer-template": true,
    "promise-function-async": false,
    "quotemark": Array [
      true,
      "single",
      "jsx-single",
    ],
    "radix": true,
    "semicolon": Array [
      true,
      "always",
    ],
    "space-before-function-paren": Array [
      true,
      "always",
    ],
    "space-within-parens": Array [
      true,
      0,
    ],
    "static-this": true,
    "switch-default": true,
    "trailing-comma": Array [
      true,
      Object {
        "esSpecCompliant": true,
        "multiline": Object {
          "arrays": "never",
          "functions": "never",
          "objects": "always",
          "typeLiterals": "ignore",
        },
        "singleline": "never",
      },
    ],
    "triple-equals": true,
    "unnecessary-else": true,
    "variable-name": Array [
      true,
      "ban-keywords",
      "check-format",
      "require-const-for-all-caps",
    ],
  },
}
`;
